src/Helpers/DownloadReport.js

```diff
@@ -0,0 +1,78 @@
+import { downloadFile } from '@redhat-cloud-services/frontend-components-utilities/files/helpers';
+
+class DownloadReport  {
+    constructor() {
+        this.defaultParams = {
+            page: 1,
+            page_size: Number.MAX_SAFE_INTEGER,
+            date: new Date().toISOString().replace(/[T:]/g, '-').split('.')[0] + '-utc'
+        };
+
+        this.params = {
+            cve: ''
+        };
+
+        this.pages = ['systems-exposed', 'system-cves', 'cves', 'system-list'];
+    }
+
+    filename = (page) => {
+        switch (page) {
+            case 'systems-exposed':
+                return `vulnerability_systems-exposed-${this.params.cve}--${this.defaultParams.date}`;
+            case 'system-list':
+                return `vulnerability_systems--${this.defaultParams.date}`;
+            default:
+                return `vulnerability_cves--${this.defaultParams.date}`;
+
+        }
+    }
+
+    checkTypes = (fetchData, page) => {
+        if (typeof fetchData !== 'function') {
+            throw `exec expects the 1st param to be funtion. Received ${typeof fetchData}`;
+        }
+
+        if (!this.pages.includes(page)) {
+            throw `Page ${page} does not exist`;
+        }
+    }
+
+    /**
+     * Execute and download the report
+     *
+     * @param {Function} fetchData
+     * @param {Object || Array} params
+     * @param {String} format csv/json
+     * @param {String} page
+     * @returns {NULL}
```

Comment:
Yes! Documentation is always good, but aside from types, this doesn't provide a lot of additional info about the function. Maybe add description what does the parameter do and what it consists of if it's an object. This is just my opinion and I don't know what depth the docu is supposed to have.

Reply:
Do you mean more description about the params? like

` @param {Function} fetchData - Function to be called in order to fetch data`
`@param {Object || Array} params - Object to be sent as request params. If you need to pass extra argument, pass an array. `

Reply:
Yes something like this, because as someone not knowing the code, I'd have no idea what format csv/json or params are.

Reply:
Maybe not as explicit like fetchData - Function to be called in order to fetch data but "format csv/json" I'd have no idea

Reply:
that's a great point. I will update it